src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.PatternLocalRewriter.cs

```diff
@@ -296,6 +296,9 @@ void addArg(RefKind refKind, BoundExpression expression)
                             return _factory.AssignmentExpression(output, access);
                         }
 
+                    case BoundDagAssignmentEvaluation:
```

Comment:
>case BoundDagAssignmentEvaluation: [](http://example.com/codeflow?start=20&length=34)

Is ```default:``` branch not good enough?

Reply:
I'd rather mention every evaluation node for an exhaustive match. The type check itself is not reachable anyways. 
(I think this also address your next comment)

Reply:
> I'd rather mention every evaluation node for an exhaustive match. The type check itself is not reachable anyways. 
>(I think this also address your next comment)

How does this help to diagnose a potential failure? It feels like ```throw ExceptionUtilities.UnexpectedValue(evaluation)``` is going to be more informative by comparison to ```throw ExceptionUtilities.Unreachable```.

Reply:
I think UnexpectedValue is useful when new nodes are added and need to be covered, while this particular node already exists and with this change it must be unreachable here.
We do have some other instances of this pattern in the codebase.

Reply:
Will change to `throw ExceptionUtilities.UnexpectedValue(evaluation.Kind);` Turned out that's more common than throwing unreachable for this propose.

Reply:
> I think UnexpectedValue is useful when new nodes are added and need to be covered, while this particular node already exists and with this change it must be unreachable here.
>We do have some other instances of this pattern in the codebase.

I think ```ExceptionUtilities.UnexpectedValue(evaluation)``` is more informative and, while investigating a failure, I wouldn't really care much about whether a new node caused a failure. I definitely wouldn't try to deduce this information from the kind of exception thrown. We use ```ExceptionUtilities.UnexpectedValue(evaluation)``` for ```default``` case here and, in my opinion, there is no good reason to diverge for ```case BoundDagAssignmentEvaluation```. We could still keep the ```case``` for it if you prefer.

Reply:
Done. I think I've addressed all the comments. ptal, thanks.